sql-plugin/src/main/scala/com/nvidia/spark/rapids/SpillableColumnarBatch.scala

```diff
@@ -158,12 +158,14 @@ object SpillableColumnarBatch {
       ct: ContiguousTable,
       sparkTypes: Array[DataType],
       priority: Long): SpillableColumnarBatch = {
-    val handle = RapidsBufferCatalog.addContiguousTable(ct, priority)
-    withResource(RapidsBufferCatalog.acquireBuffer(handle)) { _ =>
-      new SpillableColumnarBatchImpl(
-        handle,
-        ct.getRowCount.toInt,
-        sparkTypes)
+    withResource(ct) { _ =>
+      val handle = RapidsBufferCatalog.addContiguousTable(ct, priority)
+      withResource(RapidsBufferCatalog.acquireBuffer(handle)) { _ =>
+        new SpillableColumnarBatchImpl(
+          handle,
+          ct.getRowCount.toInt,
+          sparkTypes)
+      }
```

Comment:
Doesn't seem necessary to acquire the buffer since we're ignoring the result anyway?
```suggestion
      new SpillableColumnarBatchImpl(handle, ct.getRowCount.toInt, sparkTypes)
```

Reply:
will change